### PR TITLE
Refactor surface input with vector of variants

### PR DESCRIPTION
**Pull request by sethrj**
This changes the ORANGE input to use surface variants rather than raw compressed surfaces. This should be a more stable method of passing data between SCALE and ORANGE in the short term. *Marking as draft until #923 is merged and the existing SCALE interface is updated.*

### CODE REVIEW

**Comment by elliottbiondo** [comment]
As best as I can tell, this is going to break the `Celeritas_Converter` code within Geometria/orange, so maybe this should wait until @tmdelellis's branch is merged.

**Comment by sethrj** [comment]
Yes, that's what I meant by
 > Marking as draft until https://github.com/celeritas-project/celeritas/pull/923 is merged and the existing SCALE interface is updated. 

Once the various SCALE-related branches have been merged then I'll activate this one.

**Comment by sethrj** [comment]
Marking as draft until @elliottbiondo 's universe dispatch branch #973 is merged in.

**Comment by sethrj** [comment]
@elliottbiondo Now that the last of the empire paper PRs are in, we can start moving forward with celeritas orange input development. The eventual goal I have in mind is for the `UnitInput` to be an intermediate object managed during construction: so we build up and deduplicate the vector of surfaces in this container.

File: src/orange/construct/OrangeInputIO.json.cc
Comment (amandalund) [inline src/orange/construct/OrangeInputIO.json.cc]: Could you use `make_transform()` here?
```suggestion
            daughter.transform
                = detail::make_transform(slice<3>(make_span(translations), i));
```

File: test/orange/OrangeGeoTestBase.cc
Comment (amandalund) [inline test/orange/OrangeGeoTestBase.cc]: Was the change to this bounding box intentional?

**Review by amandalund — APPROVED** [review]
Looks good to me @sethrj!

File: src/orange/construct/OrangeInputIO.json.cc
Comment (sethrj) [reply src/orange/construct/OrangeInputIO.json.cc]: Yes exactly, thank you. This file has been rebased too many times 😔

File: test/orange/OrangeGeoTestBase.cc
Comment (sethrj) [reply test/orange/OrangeGeoTestBase.cc]: Nope, great catch!

**Pull request merged**
